associated-token-account/program/src/processor.rs

```diff
@@ -41,7 +43,7 @@ pub fn process_instruction(
 
     let associated_token_account_signer_seeds: &[&[_]] = &[
         &wallet_account_info.key.to_bytes(),
-        &spl_token::id().to_bytes(),
```

Comment:
So, spl-associated-program wrongly assumed `spl_token` isn't position-independent-code; but it actually IS. spent some time to debug... I suspected myself..  It turned out I needed these program changes... my js was correct. lol